### PR TITLE
ListObjectVersionsAsync replaced. VersionItem class removed. 

### PR DESCRIPTION
**Pull request by BigUstad**
Signed-off-by: Priyank Raj <priyank.raj@gatech.edu>

### CODE REVIEW

File: Minio/ApiEndpoints/BucketOperations.cs [hunk 1]
```diff
@@ -216,36 +216,30 @@ public async Task<ListAllMyBucketsResult> ListBucketsAsync(CancellationToken can
...
-            var bucketExistTask = this.BucketExistsAsync(bucketExistsArgs, cancellationToken);
-            Task.WaitAll(bucketExistTask);
-            var found = bucketExistTask.Result;
-            if (!found)
-            {
-                throw new BucketNotFoundException(args.BucketName, "Bucket not found.");
-            }
-
+            args.Validate();
+            args.Versions = true;
```
Comment (kannappanr) [inline Minio/ApiEndpoints/BucketOperations.cs:232]: why is this set to `true`?

Comment (BigUstad) [reply Minio/ApiEndpoints/BucketOperations.cs:232]: Removed

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 2]
```diff
@@ -525,64 +524,60 @@ internal async static Task TearDown(MinioClient minio, string bucketName)
...
             try
             {
+                await minio.GetVersioningAsync(new GetVersioningArgs()
+                                                            .WithBucket(bucketName));
                 lockConfig = await minio.GetObjectLockConfigurationAsync(lockConfigurationArgs);
             }
             catch (MissingObjectLockConfiguration)
             {
                 // This exception is expected for those buckets created without a lock.
+                getVersions = false;
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:541]: this probably should be rremoved

Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs]: Changed.

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**